### PR TITLE
mgr/dashboard: Allow editing iSCSI targets with initiators logged-in

### PR DESCRIPTION
**Pull request by tspmelo**
Fixes: https://tracker.ceph.com/issues/47393

Signed-off-by: Tiago Melo <tmelo@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/iscsi.py [hunk 1]
```diff
@@ -818,9 +837,8 @@ def _create(target_iqn, target_controls, acl_enabled,
                 image_ids = []
                 for disk in group['disks']:
                     image_ids.append('{}/{}'.format(disk['pool'], disk['image']))
-                if not target_config or group_id not in target_config['groups']:
-                    IscsiClient.instance(gateway_name=gateway_name).create_group(
-                        target_iqn, group_id, members, image_ids)
+                IscsiClient.instance(gateway_name=gateway_name).create_group(
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/controllers/iscsi.py:840]: Shouldn't we call `create_group` only if needed?

File: src/pybind/mgr/dashboard/controllers/iscsi.py [hunk 2]
```diff
@@ -382,37 +382,55 @@ def _delete(target_iqn, config, task_progress_begin, task_progress_end, new_targ
             task_progress_inc = int((task_progress_end - task_progress_begin) / task_progress_steps)
         gateway_name = list(target_config['portals'].keys())[0]
         deleted_groups = []
+        deleted_group_disk = []
+        deleted_group_member = []
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/controllers/iscsi.py:386]: `deleted_group_disk` and `deleted_group_member` not being used?

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by LenzGr** [comment]
Could you please fix the linting error that makes the test to fail?
```
./controllers/iscsi.py:823:49: E127 continuation line over-indented for visual indent
```

File: src/pybind/mgr/dashboard/services/iscsi_client.py [hunk 3]
```diff
@@ -205,6 +205,15 @@ def create_group(self, target_iqn, group_name, members, image_ids, request=None)
             'disks': ','.join(image_ids)
         })
 
+    @RestClient.api_put('/api/hostgroup/{target_iqn}/{group_name}')
+    def update_group(self, target_iqn, group_name, members, image_ids, request=None):
+        logger.debug("iSCSI[%s] Creating group: %s/%s", self.gateway_name, target_iqn, group_name)
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/services/iscsi_client.py:210]: Should be `Updating group` to distinguish from `create_group` call.

**Comment by ricardoasmarques** [comment]
I've removed an initiator from a group, but disks were not moved to the initiator:

remove-initiator-from-group

**Review by ricardoasmarques — APPROVED** [review]
Apart from the `make check` failure, lgtm.

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
@ricardoasmarques frontend issues will be fixed in a separate PR: https://github.com/ceph/ceph/pull/37198

**Comment by callithea** [comment]
- [pkg build](https://shaman.ceph.com/builds/ceph/wip-laura-testing-36900-37087/) [WIP]
- QA run [PENDING]

**Review by callithea — APPROVED** [review]

**Pull request merged**